### PR TITLE
EKS v3 Support

### PR DESCRIPTION
**Pull request by schottsfired**
Highlights:
* Implement "Advanced EKS Configuration" stack
* Use framework for Spot node pool, delete Spot template
* Delete parameters that are no longer supported, add/expose new ones
* Update default CloudBees Helm chart values, e.g. separate namespace for agents by default
* Deploy all resources into `default` namespace

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by schottsfired** [comment]
Thank you for https://github.com/aws-quickstart/quickstart-amazon-eks/pull/221, we pulled it in https://github.com/aws-quickstart/quickstart-cloudbees-ci/pull/31/commits/52d3273ce7ebdd61eded8b91d9bf2104c15c07b7.

Our CI is passing ✅ 
https://github.com/cloudbees/quickstart-cloudbees-ci/commits/eks-3.0

Let us know if you spot any issues with the templates @jaymccon. Looking forward to our v3 launch.

File: templates/cloudbees-core-efs.template.yaml [hunk 1]
```diff
@@ -57,14 +55,11 @@ Resources:
...
-    Type: "Custom::Helm"
-    Version: '1.0'
+    Type: "AWSQS::Kubernetes::Helm"
     Properties:
-      ServiceToken: !Ref HelmLambdaArn
-      ClusterName: !Ref ClusterName
+      ClusterID: !Ref EKSClusterName
       Chart: stable/efs-provisioner
       Name: efs-provisioner
-      Namespace: efs-provisioner
```
Comment (jaymccon) [inline templates/cloudbees-core-efs.template.yaml:67]: Is the intent that the provisiner goes into `default` namespace, I believe that is the behavior when no namespace is specified. https://github.com/aws-quickstart/quickstart-helm-resource-provider/tree/main/docs#namespace

**Review by jaymccon — CHANGES_REQUESTED** [review]
Looking good, only small nit is that we have an updated naming scheme for entrypoint templates.

`templates/cloudbees-core-master.template.yaml` should be renamed to `templates/cloudbees-core-entrypoint-new-vpc.template.yaml` and `templates/cloudbees-core-existing-vpc.template.yaml` should be `templates/cloudbees-core-entrypoint-existing-vpc.template.yaml`.

Can also switch `cloudbees-core` to `cloudbees-ci` if you'd like. 

I will ensure that the permalinks are updated in concert with publishing the updates to s3.

File: templates/cloudbees-core-efs.template.yaml (hunk 1 above)
Comment (schottsfired) [reply templates/cloudbees-core-efs.template.yaml:67]: Yes, that's the intent, and it works exactly as you described.

**Comment by schottsfired** [comment]
Thanks for your review @jaymccon. I'm glad you brought up the rename because I noticed it, but decided not to change (read: break) anything because of the permalinks in the deployment guide. It's great to hear that you'll take care of the permalinks, so I'll go ahead with the rename now.

**Review by jaymccon — APPROVED** [review]

**Pull request merged**